pipeline/src/main/java/io/datacater/Pipeline.java

```diff
@@ -110,17 +141,42 @@ private void logMessage(String message){
     LOGGER.error(errorMsg);
   }
 
-  private JsonObject getKey(JsonObject message){
-    if(message.getJsonObject(PipelineConfig.KEY) == null){
-      return null;
+  private Deserializer getDeserializerInstance(String className) throws TransformationException {
```

Comment:
It might make sense to map out all of our Deserializers via an Enum.

Reply:
At the moment, we allow users to configure any deserializer class.

From my understanding, using an enum would require us to know potential deserializers beforehand.

Are we willing to step into this direction and decrease the flexibility? Happy to make the chance, if I get a yes :)

Reply:
In 2af7b3cc98887f1e60afade8724268585d74ce1f, I replaced the usage of the reflection API with a static Map.